controllers/clustertopology_controller_compute.go

```diff
@@ -140,9 +141,129 @@ func (r *ClusterTopologyReconciler) getClass(ctx context.Context, cluster *clust
 	return class, nil
 }
 
-// getTemplate gets the object referenced in ref.
+// getCurrentState gets information about the current state of a cluster by inspecting the state of the cluster infrastructure,
+// the control plane, and the machine deployments associated with the cluster.
+func (r *ClusterTopologyReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class clusterv1.ClusterClass) (*clusterTopologyState, error) {
+	var err error
+	var infra *unstructured.Unstructured
+	var cp controlPlaneTopologyState
+	var m []machineDeploymentTopologyState
+	// Reference to the cluster infrastructure can be nil and is expected to be on the first reconcile loop
+	// In this case the method should still be allowed to continue
+	if cluster.Spec.InfrastructureRef != nil {
+		infra, err = r.getCurrentInfrastructureClusterState(ctx, cluster)
+		if err != nil {
+			return nil, errors.Wrap(err, "current cluster state not found")
+		}
+	}
+	// Reference to the control plane can be nil, and is expected to be on the first reconcile. In this case the method
+	// should still be allowed to continue
+	if cluster.Spec.ControlPlaneRef != nil {
+		cp, err = r.getControlPlaneState(ctx, cluster, class)
+		if err != nil {
+			return nil, errors.Wrap(err, "current cluster state not found")
+		}
+	}
+	// A cluster may have zero or more machine deployments and a cluster is expected to have zero machine deployments on
+	// first reconcile
+	m, err = r.getCurrentMachineDeploymentState(ctx, cluster)
+	if err != nil {
+		return nil, errors.Wrap(err, "current cluster state not found")
+	}
+	return &clusterTopologyState{
+		cluster:               cluster,
+		infrastructureCluster: infra,
+		machineDeployments:    m,
+		controlPlane:          cp,
+	}, nil
+}
+
+// getCurrentInfrastructureClusterState looks for the infrastructure underlying the cluster. If a reference is set but not
+// found, through an error or not existing in the api server an error is thrown.
+func (r *ClusterTopologyReconciler) getCurrentInfrastructureClusterState(ctx context.Context, cluster *clusterv1.Cluster) (*unstructured.Unstructured, error) {
+	infra, err := r.getReference(ctx, cluster.Spec.InfrastructureRef)
+	if err != nil {
+		return nil, errors.Wrap(err, "cluster infrastructure state")
+	}
+	return infra, nil
+}
+
+// getControlPlaneState returns information on the control plane being used by the cluster; If a reference not found,
```

Comment:
```suggestion
// getControlPlaneState returns information on the control plane being used by the cluster. If a reference is not found,
```
(not sure about the Semikolon, your decision)

Reply:
I'll leave the semicolon  - it's not wrong, but it tends to add ambiguity in some cases